pkg/cloud/cloud.go

```diff
@@ -227,15 +230,25 @@ var _ Cloud = &cloud{}
 
 // NewCloud returns a new instance of AWS cloud
 // It panics if session is invalid
-func NewCloud(region string) (Cloud, error) {
-	return newEC2Cloud(region)
+func NewCloud(region string, assumeRoleArn string) (Cloud, error) {
+	return newEC2Cloud(region, assumeRoleArn)
 }
 
-func newEC2Cloud(region string) (Cloud, error) {
+func newEC2Cloud(region string, assumeRoleArn string) (Cloud, error) {
 	awsConfig := &aws.Config{
 		Region:                        aws.String(region),
 		CredentialsChainVerboseErrors: aws.Bool(true),
 	}
+	sess := session.Must(session.NewSession(awsConfig))
+
+	if assumeRoleArn != "" {
+		if arn.IsARN(assumeRoleArn) {
+			klog.V(5).Infof("Assuming role into ARN %s", assumeRoleArn)
+			awsConfig.Credentials = stscreds.NewCredentials(sess, assumeRoleArn)
+		} else {
```

Comment:
Can we add unit test for this check?

Reply:
added something, this one could be a bad idea because it does actually try to create a ec2 session